packages/editor/src/lib/app/App.ts

```diff
@@ -176,28 +189,43 @@ export function isShapeWithHandles(shape: TLShape) {
 
 /** @public */
 export class App extends EventEmitter<TLEventMap> {
-	constructor({ config, store, getContainer }: AppOptions) {
+	constructor({ store, user, tools = defaultTools, shapes = {}, getContainer }: AppOptions) {
 		super()
 
-		this.config = config
-
-		if (store.schema !== this.config.storeSchema) {
-			throw new Error('Store schema does not match schema given to App')
-		}
-
 		this.store = store
 
-		this.user = new UserPreferencesManager(this)
+		this.user = new UserPreferencesManager(user ?? createTldrawEditorUser())
 
 		this.getContainer = getContainer ?? (() => document.body)
 
 		this.textMeasure = new TextManager(this)
 
-		// Set the shape utils
+		for (const key in shapes) {
+			if (key in defaultShapes) {
+				throw Error(`Can't override default shape ${key}!`)
+			}
+		}
+
 		this.shapeUtils = Object.fromEntries(
-			Object.entries(this.config.shapeUtils).map(([type, Util]) => [type, new Util(this, type)])
+			Object.entries({
+				// we definitely need a group shape, god help you if you override it
+				...defaultShapes,
+				...shapes,
+			}).map(([type, { util: Util }]) => {
+				if (type !== Util.type) throw Error()
+				return [Util.type, new Util(this, Util.type)]
+			})
 		)
```

Comment:
we should def follow this up with some shape-removal stuff. i think that's maybe the next thing i'd like us to tackle after styling

Reply:
Making a small improvement here (core vs default shapes).